### ISSUE TITLE
False Negative when Enforcing Positional Arguments in regards to Protocols

### ISSUE THREAD

**Issue by max-muoto** [python/mypy#17421]
**Bug Report**

MyPy doesn't raise a warning if you pass in an class implementation that enforces positional arguments for a method for a protocol that does not.

(A clear and concise description of what the bug is.)

**To Reproduce**

```python
# Ideally, a small sample program that demonstrates the problem.
# Or even better, a reproducible playground link https://mypy-play.net/ (use the "Gist" button)
```

Example of a false negative. Executing this code leads to a runtime error, as `ActualReadable.read` will raise an error when trying to pass in a key-word argument.

```python
from typing import Protocol


class Readable(Protocol):
    def read(self, n: int = -1) -> bytes: ...

class ActualReadable:
    def read(self, n: int = -1, /) -> bytes:
        return b""


def foo(readable: Readable) -> None:
    readable.read(n=1)


foo(ActualReadable())
```

[MyPy Playground
](https://mypy-play.net/?mypy=latest&python=3.12&flags=strict)

**Expected Behavior**

Pyright's behavior would be good to model off of in terms of expected behavior. That is, raising a warning that the implementation isn't compatible with the protocol, and that `n` must not be positional-only.

[Pyright Playground
](https://pyright-play.net/?pyrightVersion=1.1.368&strict=true&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAK4MYAxmADYCwAULfXY6RQIYDObUASgKYsAmLAEYUeACiJgS5CgEoAXLSjKo-HsCgg%2B-MWx4VgAGigp5mFPgC8UALQBGWbYB8UIXBg82ZgHS%2BGzdk4AQVIYAFcWCl4BYVFFGhVVdU1tXX0jEzNUK1s7YwB6RxsXNw8vJUSVLXCQFFcAInqGZpo1DWAwMDEtGJEeM2jBPqKXADkwFH6oX28Wjq6Q8MjB2PFZWVogA)

**Your Environment**

- Mypy version used: 1.10.0
- Mypy command-line flags: Strict
- Python version used: 3.12
